pkg/generate/app/cmd/newapp.go

```diff
@@ -315,17 +316,13 @@ func (c *AppConfig) componentsForRepos(repositories app.SourceRepositories) (app
 			errs = append(errs, fmt.Errorf("source not detected for repository %q", repo))
 			continue
 		case info.Dockerfile != nil && (len(c.Strategy) == 0 || c.Strategy == "docker"):
-			df, err := info.Dockerfile.Dockerfile()
-			if err != nil {
-				// an unparseable docker file should not terminate the entire create - it may be valid to Docker but not to us
-				fmt.Fprintf(c.ErrOut, "WARNING: The Dockerfile for the repository %q could not be parsed - no image stream can be created for the FROM\n", info.Path)
+			node := info.Dockerfile.AST()
+			baseImage := dockerfileutil.LastBaseImage(node)
+			if baseImage == "" {
+				errs = append(errs, fmt.Errorf("the Dockerfile in the repository %q has no FROM instruction", info.Path))
 				continue
```

Comment:
@csrwng @bparees could you please have a closer look here?
Existing code will proceed to call `b.AddComponents` in case you have a Dockerfile like this:

```Dockerfile
# missing image argument below
FROM
RUN yum install -y apache
...
```

What should we be doing here? Maybe we should remove the `continue` and call `b.AddComponents` passing `[]string{""}`?!